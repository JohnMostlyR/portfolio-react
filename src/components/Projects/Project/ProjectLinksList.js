import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import ProjectExternalLink from './ProjectExternalLink';

const UL = styled.ul`
  height: 1.5em;
  margin: 0;
  padding: 0;
  list-style: none;
  text-align: right;
`;

const ProjectLinksList = (props) => (
  <UL>
    {
      props.links.map(
        (link) => (
          <ProjectExternalLink
            key={link.name}
            name={link.name}
            url={link.url}
            icon={link.icon}
          />
        )
      )
    }
  </UL>
);

ProjectLinksList.propTypes = {
  title: PropTypes.string,
  links: PropTypes.array,
};

export default ProjectLinksList;