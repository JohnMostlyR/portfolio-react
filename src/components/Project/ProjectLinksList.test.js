import React from 'react';
import {shallow} from 'enzyme';

import ProjectLinksList from './ProjectLinksList';

describe('ProjectLinksList', () => {
  const links = [
    {
      name: 'github',
      url: 'https://github.com/',
      faIcon: 'github',
    },
    {
      name: 'codepen',
      url: 'https://codepen.io/',
      faIcon: 'codepen',
    },
  ];

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ProjectLinksList links={links}/>);
  });

  it('should render and match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});