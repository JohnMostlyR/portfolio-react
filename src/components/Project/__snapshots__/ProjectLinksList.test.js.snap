// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectLinksList should render and match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProjectLinksList
    links={
        Array [
            Object {
              "faIcon": "github",
              "name": "github",
              "url": "https://github.com/",
            },
            Object {
              "faIcon": "codepen",
              "name": "codepen",
              "url": "https://codepen.io/",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <ProjectExternalLink
          faIcon="github"
          name="github"
          url="https://github.com/"
/>,
        <ProjectExternalLink
          faIcon="codepen"
          name="codepen"
          url="https://codepen.io/"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "faIcon": "github",
          "name": "github",
          "url": "https://github.com/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "faIcon": "codepen",
          "name": "codepen",
          "url": "https://codepen.io/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <ProjectExternalLink
            faIcon="github"
            name="github"
            url="https://github.com/"
/>,
          <ProjectExternalLink
            faIcon="codepen"
            name="codepen"
            url="https://codepen.io/"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "faIcon": "github",
            "name": "github",
            "url": "https://github.com/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "faIcon": "codepen",
            "name": "codepen",
            "url": "https://codepen.io/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
